src/shared/tests.h

```diff
@@ -201,41 +201,47 @@ static inline int run_test_table(void) {
-                        abort();                                                                         \
-                }                                                                                        \
+#define ASSERT_EQ(expr1, expr2)                                                                                  \
+        ({                                                                                                       \
+                typeof(expr1) _expr1 = (expr1);                                                                  \
+                typeof(expr2) _expr2 = (expr2);                                                                  \
+                if (_expr1 != _expr2) {                                                                          \
+                        log_error("Assertion failed in file %s at line %i: "                                     \
+                                  "expected %s == %s, but %d != %d",                                             \
+                                  PROJECT_FILE, __LINE__, #expr1, #expr2, _expr1, _expr2);                       \
+                        abort();                                                                                 \
+                }                                                                                                \
         })
 
-#define ASSERT_GE(expr1, expr2)                                                                          \
-        ({                                                                                               \
-                int _expr1 = (expr1);                                                                    \
-                int _expr2 = (expr2);                                                                    \
-                if (_expr1 < _expr2) {                                                                   \
-                        log_error("Assertion failed: expected %s >= %s, but %d < %d", #expr1, #expr2, _expr1, _expr2); \
-                        abort();                                                                         \
-                }                                                                                        \
+#define ASSERT_GE(expr1, expr2)                                                                                  \
+        ({                                                                                                       \
+                typeof(expr1) _expr1 = (expr1);                                                                  \
+                typeof(expr2) _expr2 = (expr2);                                                                  \
+                if (_expr1 < _expr2) {                                                                           \
+                        log_error("Assertion failed in file %s at line %i: "                                     \
```

Comment:
Please format the line and file here as you do above (and do the same for all the other macros you added as well)